### DISCUSSION TITLE
postfix/smtpd[409588]: warning: connect to Milter service inet:localhost:8891: Connection refused

### DISCUSSION THREAD

**Discussion by pontus14553** [docker-mailserver/docker-mailserver#2600] [Q&A]
Anyone knows why im getting this error?
postfix/smtpd[409588]: warning: connect to Milter service inet:localhost:8891: Connection refused

**Comment by mholti** [comment]
Hi, 
the dkim-milter is listeining on port TCP/8891. As long, as you don't have dkim configured you can ignore this error as far as i know...

**Comment by andrewlow** [comment]
Maybe I missed this previously - but I think this is new error in 11.0. I almost logged a bug, but it's not stopping anything from working - still seems like an error that could be fixed with the right configuration.

**Reply by andrewlow** [reply]
I may have to take that back.. looking inside my mail container
```
root     10535  2.0  0.0      0     0 ?        Z    08:32   0:00 [opendkim] <defunct>
```
that seems.. bad..

**Reply by andrewlow** [reply]
Maybe related to the start up log
```
[ WARNING ]  No DKIM key(s) provided - check the documentation on how to get your keys
```

I'm trying to recall if I had set this up or not..

**Reply by andrewlow** [reply]
Ok - can confirm I have NOT setup DKIM - I rely on Amazon SES to wrap outgoing messages in DKIM. 

However, the miter message is about incoming email - so I'm not sure what's up here.

**Reply by andrewlow** [reply]
I'm starting to think - because I opted to NOT setup DKIM - I have no config file - and thus DKIM isn't active in either direction. This is causing the error on incoming, despite my not strictly needing outgoing.

**Comment by andrewlow** [comment]
I can confirm - setting up DKIM resolve the error in the log about mitre connection

https://docker-mailserver.github.io/docker-mailserver/edge/config/best-practices/dkim/

I opted to use a 2048 bit key. The mitre errors were non-fatal, email was flowing fine previously.

**Reply by andrewlow** [reply]
This is a useful tool to verify that email is setup correct etc.. https://mxtoolbox.com/deliverability

**Comment by pontus14553** [comment]
Thank you, that was the problem! Adding a DKIM key make the daemon start :-) Incoming DKIM is verified, and outgoing emails have DKIM.. Perfect and thanks again!
